packages/autocomplete-plugin-redirect-url/src/types/Redirect.ts

```diff
@@ -24,18 +24,54 @@ export type OnRedirectOptions<TItem extends RedirectUrlItem> = {
   state: AutocompleteState<TItem>;
 };
 
-export type TransformResponseParams<TItem> =
+/**
+ * Used to map response data into values required by the plugin.
+ */
+export interface TransformResponse {
+  /**
+   * The query used in the request.
+   *
+   * This is needed to accurately match the redirect with the correct query.
+   * Otherwise, there is a risk of the last request reflecting a different query value.
+   * (ex: typing "shoes" quickly risks mismatching the redirect item with "sho" instead)
+   */
+  queryUsed: string | undefined;
```

Comment:
it's annoying but for now you'll need to add a different function to `transformResponseToQuery` or something, otherwise it's a breaking change